resources/resource_factories/create/create.go

```diff
@@ -126,3 +137,137 @@ func (c *Client) FromString(targetPath, content string) (resource.Resource, erro
 			})
 	})
 }
+
+// FromRemote expects one or n-parts of a URL to a resource
+// If you provide multiple parts they will be joined together to the final URL.
+func (c *Client) FromRemote(args ...interface{}) (resource.Resource, error) {
+	uri, headers := toURLAndHeaders(args)
+	rURL, err := url.Parse(uri)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to parse URL for resource %s", uri)
+	}
+
+	resourceID := helpers.MD5String(uri)
+
+	return c.rs.ResourceCache.GetOrCreate(path.Join(resources.CACHE_OTHER, resourceID), func() (resource.Resource, error) {
```

Comment:
There is a long story, but I'm currently on a branch that will improve this "dependency tracking vs cache" situation (also, but not now (we should mention it in the documentation), this only caches to memory, which may not be optimal for lots of remote resources).

But, using `resources.CACHE_OTHER` will effectively trigger a re-download of all remote resources on every rebuild, which is probably not what you would want. I suggest you use `resources.ResourceCacheKey`).

Reply:
871bbd4 changes this to `resources.ResourceCacheKey`, does this mean a cache of the remote resource is stored to disk and retained between builds?

As it only returns a string, I doubt it, but I haven't researched the cache system.

Reply:
No, but the `resources.ResourceCacheKey` prevents it from being re-downloaded on every rebuild in server mode. There is a long story here; I'm doing some major cleanups in this area in another branch, and we will improve this situation before we release Hugo 0.90, so just let it be for now.